### DISCUSSION TITLE
vllm API- add regex support

### DISCUSSION THREAD

**Discussion by mjsteele12** [outlines-dev/outlines#474] [Q&A] [answered]
The [JSON implementation](https://outlines-dev.github.io/outlines/reference/vllm/) for vllm API is fantastic! I am opening this discussion to suggest adding regex support since I believe it will be equally useful. IMO one place where local llms really shine is running data ETL/NLP tasks around the clock. This requires guaranteed output structure provided by regex/outlines. For example, something I am working on right now in my RAG pipeline is to add structured/curated metadata derived from text which can be used to drastically reduce the search space. Regex support is crucial for extraction tasks like dates, emails, phone numbers etc.

**Comment by rlouf** [comment, marked as answer]
Implemented by #481
